alpha/apps/kaltura/lib/kUploadTokenMgr.php

```diff
@@ -117,6 +117,24 @@ public function getUploadToken()
 	 */
 	public function uploadFileToToken($fileData, $resume = false, $resumeAt = -1)
 	{
+		if (!$this->_uploadToken->getFileName())
+		{
+			$this->_uploadToken->setFileName($fileData['name']);
+		}
+		$entry = $this->getAttachedEntry();
+		if ($entry && $entry->getType() == entryType::MEDIA_CLIP && !$entry->getMediaType())
+		{
+			$entry->setMediaType($this->getMediaType());
```

Comment:
change it to 2 phases get and then set so the entry wont indexed if we dont have to,
like we have on:
api_v3/lib/KalturaEntryService.php:444